Code Audit Routines — Open: 0 | Closed: 1464 | Simplified: | Metric | Count |

## Code Audit Routines

**Last sweep**: `2026-06-09T22:53:34Z`
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5
**Badge status**: 

### Quality

| Metric | Value |
| --- | --- |
| SonarCloud gate | OK |
| SonarCloud issues | 0 (0 BLOCKER · 0 CRITICAL · 0 MAJOR · 0 MINOR · 0 INFO) |
| Qlty grade (local, from smell count) | UNKNOWN |
| Qlty smells | 0 → 0 (unchanged) |

### Simplification

| Metric | Value |
| --- | --- |
| Open | 0 |
| Closed | 1464 |
| Simplified (state tracked) | | Metric | Count | |
| Resolution rate | 100% |

### PR Scanning

| Metric | Value |
| --- | --- |
| PRs scanned (lifetime) | 87 |
| Issues created (lifetime) | 0 |

### Bot Review Coverage

---
_Auto-updated by daily quality sweep. Findings as of `2026-06-09T22:53:34Z` — may not reflect recent merges between sweeps._
_The codebase is the primary source of truth. This dashboard is a reporting snapshot that may lag behind reality._
_Do not edit manually._

## Daily Code Quality Sweep

**Date**: 2026-04-29T05:10:40Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 40
- **Notes**: 94

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/aidevops-update-check.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-chase-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: dataset_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: model_args appears unused. Verify use (or export if used externally). [SC2034]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: ERROR
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 3 (ERROR)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

**Failing conditions (badge blockers):**
- **new_security_rating**: actual=3, required GT 1 -- Fix new vulnerabilities introduced in the analysis period
- **Open issues**: 59
- **By severity**:
  - MAJOR: 59
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - VULNERABILITY: 58
  - CODE_SMELL: 1
  - BUG: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S8541: 58 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: quality gate ERROR, issue spike +58

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.10 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-30T06:00:02Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 4

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 37
- **Notes**: 94

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-chase-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: ERROR
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 3 (ERROR)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

**Failing conditions (badge blockers):**
- **new_security_rating**: actual=3, required GT 1 -- Fix new vulnerabilities introduced in the analysis period
- **Open issues**: 64
- **By severity**:
  - MAJOR: 64
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - VULNERABILITY: 63
  - CODE_SMELL: 1
  - BUG: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S8541: 63 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: quality gate ERROR

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy


---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.12 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-01T20:01:20Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 37
- **Notes**: 80

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-stashes-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-stashes-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: ERROR
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 3 (ERROR)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

**Failing conditions (badge blockers):**
- **new_security_rating**: actual=3, required GT 1 -- Fix new vulnerabilities introduced in the analysis period
- **Open issues**: 64
- **By severity**:
  - MAJOR: 64
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - VULNERABILITY: 63
  - CODE_SMELL: 1
  - BUG: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S8541: 63 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: quality gate ERROR

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 3
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.89 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-05T00:36:16Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 37
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: ERROR
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 0.0 (ERROR)

**Failing conditions (badge blockers):**
- **new_security_hotspots_reviewed**: actual=0.0, required LT 100 -- Review unreviewed security hotspots in SonarCloud UI (mark Safe/Fixed) or fix the flagged code

**Unreviewed security hotspots (1):**
  - .agents/scripts/shared-gh-wrappers-status.sh:97 — shell:S4790: Make sure this weak hash algorithm is not used in a sensitive context here.
_Review these in SonarCloud UI or fix the underlying code to pass the quality gate._
- **Open issues**: 1
- **By severity**:
  - MAJOR: 1
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 1
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: quality gate ERROR

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 11
- **Findings**: 1
- **Issues created**: 1

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.14.57 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-09T04:26:38Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 15
- **Findings**: 16
- **Issues created**: 8

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.15.9 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-10T05:02:38Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 19
- **Findings**: 41
- **Issues created**: 15

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.15.25 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-12T06:52:33Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 94

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 16
- **Findings**: 19
- **Issues created**: 6

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.15.34 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-15T20:23:08Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 107

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-chase-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 13
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.15.54 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-24T23:27:25Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 14
- **Findings**: 28
- **Issues created**: 9

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.17.30 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-25T23:56:35Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 18
- **Findings**: 50
- **Issues created**: 17

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.19.0 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-27T01:08:32Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 29
- **Findings**: 40
- **Issues created**: 13

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.19.5 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-05-30T17:28:45Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 108

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/badges-check-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-chase-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-draft-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/check-workflows-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 3
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.20.5 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-06-01T06:00:47Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 0
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.20.7 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-06-06T07:54:19Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 38
- **Notes**: 95

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper-commands.sh`: header_file appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_etag appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper-import.sh`: resp_last_modified appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: SKILL_SOURCES appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/add-skill-helper.sh`: TEMP_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-cli/aidevops-init-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-repos-lib.sh`: REPOS_FILE may not be assigned. Did you mean repos_file? [SC2153]
  - `.agents/scripts/aidevops-cli/aidevops-skills-plugin-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-status-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-cli/aidevops-update-lib.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/anti-detect-helper.sh`: PROFILES_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/anti-detect-helper.sh`: VENV_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper-scheduler.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-helper.sh`: SCREENSHOTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: QA_RESULTS_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-asset-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaign-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-meta-filer.sh`: ./shared-gh-wrappers.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: OFFLINE_OFFSET appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 0
- **By severity**:
  - INFO: 0
  - MINOR: 0
  - MAJOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 0
  - BUG: 0
  - VULNERABILITY: 0

### CodeRabbit

_Monitoring: 0 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 0
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.20.17 automated scan.

--- a/.agents/scripts/shared-repo-state-guard.sh
+++ b/.agents/scripts/shared-repo-state-guard.sh
@@ -75,3 +75,65 @@ aidevops_can_manage_repo_issue_state() {
 
 	return 1
 }
+
+#######################################
+# Check whether aidevops may run public repo routines for a repo.
+#
+# Routine jobs (quality sweeps, health dashboards, merge/repair loops) can
+# create comments/issues and consume API budget. Unlike narrow issue-state
+# markers, these jobs require maintainer-equivalent authority; write-only
+# collaborators and external contributors should keep results local.
+#
+# Args:
+#   $1 = repo slug (owner/repo)
+#   $2 = optional authenticated login override
+# Returns: 0 when routines are allowed, 1 otherwise.
+#######################################
+aidevops_can_run_repo_routines() {
+	local slug="${1:-}"
+	local user="${2:-}"
+
+	if [[ "${AIDEVOPS_TEST_MODE:-}" == "1" && \
+		"${AIDEVOPS_REPO_ROUTINE_GUARD_TEST_BYPASS:-}" == "1" ]]; then
+		return 0
+	fi
+
+	if [[ -z "$slug" || "$slug" != */* ]]; then
+		return 1
+	fi
+
+	if [[ -z "$user" ]]; then
+		user=$(aidevops_repo_state_current_user)
+	fi
+	if [[ -z "$user" ]]; then
+		return 1
+	fi
+
+	local repo_owner="${slug%%/*}"
+	if [[ "$user" == "$repo_owner" ]]; then
+		return 0
+	fi
+
+	local repos_json="${REPOS_JSON:-${HOME}/.config/aidevops/repos.json}"
+	if [[ -f "$repos_json" ]]; then
+		local registered_maintainer=""
+		registered_maintainer=$(jq -r \
+			--arg slug "$slug" \
+			'.initialized_repos[] | select(.slug == $slug) | .maintainer // empty' \
+			"$repos_json" 2>/dev/null | sed -n '1p')
+		if [[ -n "$registered_maintainer" && "$registered_maintainer" == "$user" ]]; then
+			return 0
+		fi
+	fi
+
+	local permission=""
+	permission=$(gh api "repos/${slug}/collaborators/${user}/permission" \
+		--jq '.permission // ""' 2>/dev/null || printf '')
+	case "$permission" in
+		admin | maintain)
+			return 0
+			;;
+	esac
+
+	return 1
+}

--- a/.agents/scripts/stats-functions.sh
+++ b/.agents/scripts/stats-functions.sh
@@ -57,6 +57,9 @@ fi
 STATS_SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]:-$0}")" && pwd)"
 
 # Phase 1: shared utility functions (slug validation, runner role)
+# shellcheck source=shared-repo-state-guard.sh
+source "${STATS_SCRIPT_DIR}/shared-repo-state-guard.sh"
+
 # shellcheck source=stats-shared.sh
 source "${STATS_SCRIPT_DIR}/stats-shared.sh"
 

--- a/.agents/scripts/stats-health-dashboard.sh
+++ b/.agents/scripts/stats-health-dashboard.sh
@@ -267,6 +267,29 @@ _update_health_issue_for_repo() {
 }
 
 #######################################
+# Filter repo entries to repos where public routines are authorized.
+# Arguments:
+#   $1 - newline-delimited slug|path entries
+#   $2 - authenticated GitHub user
+# Output: authorized slug|path entries
+#######################################
+_filter_routine_eligible_repo_entries() {
+	local repo_entries="$1"
+	local routine_runner_user="$2"
+	local slug path
+
+	while IFS='|' read -r slug path; do
+		[[ -z "$slug" ]] && continue
+		if ! aidevops_can_run_repo_routines "$slug" "$routine_runner_user"; then
+			echo "[stats] Health dashboard skipped for ${slug}: ${routine_runner_user} is not maintainer-equivalent" >>"$LOGFILE"
+			continue
+		fi
+		printf '%s|%s\n' "$slug" "$path"
+	done <<<"$repo_entries"
+	return 0
+}
+
+#######################################
 # Update health issues for ALL pulse-enabled repos
 #
 # Iterates repos.json and calls _update_health_issue_for_repo for each
@@ -296,6 +319,18 @@ update_health_issues() {
 		return 0
 	fi
 
+	local routine_runner_user
+	routine_runner_user=$(aidevops_repo_state_current_user)
+	if [[ -z "$routine_runner_user" ]]; then
+		echo "[stats] Health dashboard skipped: could not resolve authenticated GitHub user" >>"$LOGFILE"
+		return 0
+	fi
+
+	repo_entries=$(_filter_routine_eligible_repo_entries "$repo_entries" "$routine_runner_user")
+	if [[ -z "$repo_entries" ]]; then
+		return 0
+	fi
+
 	# Refresh person-stats cache if stale (t1426: hourly, not every pulse)
 	_refresh_person_stats_cache || true
 
@@ -312,7 +347,7 @@ update_health_issues() {
 	local activity_helper="${HOME}/.aidevops/agents/scripts/contributor-activity-helper.sh"
 	if [[ -x "$activity_helper" ]]; then
 		local all_repo_paths
-		all_repo_paths=$(jq -r '.initialized_repos[] | select(.pulse == true and (.local_only // false) == false) | .path' "$repos_json" || echo "")
+		all_repo_paths=$(printf '%s\n' "$repo_entries" | awk -F'|' 'NF >= 2 && $2 != "" { print $2 }')
 		if [[ -n "$all_repo_paths" ]]; then
 			local -a cross_args=()
 			while IFS= read -r rp; do

--- a/.agents/scripts/stats-quality-sweep.sh
+++ b/.agents/scripts/stats-quality-sweep.sh
@@ -143,12 +143,23 @@ run_daily_quality_sweep() {
 		return 0
 	fi
 
+	local routine_runner_user
+	routine_runner_user=$(aidevops_repo_state_current_user)
+	if [[ -z "$routine_runner_user" ]]; then
+		echo "[stats] Quality sweep skipped: could not resolve authenticated GitHub user" >>"$LOGFILE"
+		return 0
+	fi
+
 	echo "[stats] Starting daily code quality sweep..." >>"$LOGFILE"
 
 	local swept=0
 	while IFS='|' read -r slug path; do
 		[[ -z "$slug" ]] && continue
 		[[ ! -d "$path" ]] && continue
+		if ! aidevops_can_run_repo_routines "$slug" "$routine_runner_user"; then
+			echo "[stats] Quality sweep skipped for ${slug}: ${routine_runner_user} is not maintainer-equivalent" >>"$LOGFILE"
+			continue
+		fi
 		_quality_sweep_for_repo "$slug" "$path" || true
 		swept=$((swept + 1))
 	done <<<"$repo_entries"

--- a/.agents/scripts/tests/test-repo-state-guard.sh
+++ b/.agents/scripts/tests/test-repo-state-guard.sh
@@ -29,6 +29,10 @@ if [[ "$1" == "api" && "$2" == repos/*/collaborators/*/permission ]]; then
     printf '%s\n' "write"
     exit 0
   fi
+  if [[ "$mode" == "routine-maintain" ]]; then
+    printf '%s\n' "maintain"
+    exit 0
+  fi
   printf '%s\n' '{"message":"Must have push access"}' >&2
   exit 1
 fi
@@ -77,10 +81,22 @@ else
 	check 0 "repo owner may manage state"
 fi
 
+if aidevops_can_run_repo_routines "tester/project"; then
+	check 1 "repo owner may run routines"
+else
+	check 0 "repo owner may run routines"
+fi
+
 if aidevops_can_manage_repo_issue_state "other/project"; then
 	check 0 "external non-maintainer is blocked"
 else
 	check 1 "external non-maintainer is blocked"
+fi
+
+if aidevops_can_run_repo_routines "other/project"; then
+	check 0 "external non-maintainer routine is blocked"
+else
+	check 1 "external non-maintainer routine is blocked"
 fi
 
 printf 'managed\n' >"$mode_file"
@@ -88,6 +104,19 @@ if aidevops_can_manage_repo_issue_state "other/project"; then
 	check 1 "write collaborator may manage state"
 else
 	check 0 "write collaborator may manage state"
+fi
+
+if aidevops_can_run_repo_routines "other/project"; then
+	check 0 "write collaborator routine is blocked"
+else
+	check 1 "write collaborator routine is blocked"
+fi
+
+printf 'routine-maintain\n' >"$mode_file"
+if aidevops_can_run_repo_routines "other/project"; then
+	check 1 "maintain collaborator may run routines"
+else
+	check 0 "maintain collaborator may run routines"
 fi
 
 printf 'external\n' >"$mode_file"

--- a/.agents/scripts/tests/test-stats-routine-maintainer-guard.sh
+++ b/.agents/scripts/tests/test-stats-routine-maintainer-guard.sh
@@ -1,0 +1,103 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2025-2026 Marcus Quinn
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+AGENTS_SCRIPTS_DIR="$(cd "${SCRIPT_DIR}/.." && pwd)"
+
+tmp_dir=$(mktemp -d)
+trap 'rm -rf "$tmp_dir"' EXIT
+
+mkdir -p "${tmp_dir}/bin" "${tmp_dir}/repo-write" "${tmp_dir}/repo-maintain" "${tmp_dir}/logs"
+
+cat >"${tmp_dir}/bin/gh" <<'STUB'
+#!/usr/bin/env bash
+set -euo pipefail
+
+if [[ "$1" == "auth" && "$2" == "status" ]]; then
+	exit 0
+fi
+
+if [[ "$1" == "api" && "$2" == "user" ]]; then
+	printf '%s\n' 'tester'
+	exit 0
+fi
+
+if [[ "$1" == "api" && "$2" == repos/*/collaborators/tester/permission ]]; then
+	case "$2" in
+		repos/other/write/collaborators/tester/permission)
+			printf '%s\n' 'write'
+			;;
+		repos/other/maintain/collaborators/tester/permission)
+			printf '%s\n' 'maintain'
+			;;
+		*)
+			printf '%s\n' '{"message":"not found"}' >&2
+			exit 1
+			;;
+	esac
+	exit 0
+fi
+
+exit 0
+STUB
+chmod +x "${tmp_dir}/bin/gh"
+
+cat >"${tmp_dir}/repos.json" <<JSON
+{
+  "initialized_repos": [
+    {"slug": "other/write", "path": "${tmp_dir}/repo-write", "pulse": true},
+    {"slug": "other/maintain", "path": "${tmp_dir}/repo-maintain", "pulse": true}
+  ]
+}
+JSON
+
+export PATH="${tmp_dir}/bin:${PATH}"
+export HOME="$tmp_dir"
+export REPOS_JSON="${tmp_dir}/repos.json"
+export LOGFILE="${tmp_dir}/logs/stats.log"
+export QUALITY_SWEEP_INTERVAL=0
+export QUALITY_SWEEP_LAST_RUN="${tmp_dir}/logs/quality-sweep-last-run"
+export PERSON_STATS_LAST_RUN="${tmp_dir}/logs/person-stats-last-run"
+export PERSON_STATS_CACHE_DIR="${tmp_dir}/logs"
+export QUALITY_SWEEP_STATE_DIR="${tmp_dir}/logs/quality-sweep-state"
+
+# stats-functions.sh submodules expect caller SCRIPT_DIR to be the scripts dir,
+# matching stats-wrapper.sh production sourcing.
+SCRIPT_DIR="$AGENTS_SCRIPTS_DIR"
+
+# shellcheck source=../shared-constants.sh
+source "${AGENTS_SCRIPTS_DIR}/shared-constants.sh"
+# shellcheck source=../worker-lifecycle-common.sh
+source "${AGENTS_SCRIPTS_DIR}/worker-lifecycle-common.sh"
+# shellcheck source=../stats-functions.sh
+source "${AGENTS_SCRIPTS_DIR}/stats-functions.sh"
+
+called_file="${tmp_dir}/called"
+_quality_sweep_for_repo() {
+	local repo_slug="$1"
+	local repo_path="$2"
+	printf '%s|%s\n' "$repo_slug" "$repo_path" >>"$called_file"
+	return 0
+}
+
+run_daily_quality_sweep
+
+if [[ ! -f "$called_file" ]]; then
+	printf 'FAIL no eligible repo was swept\n'
+	exit 1
+fi
+
+if grep -q '^other/write|' "$called_file"; then
+	printf 'FAIL write-only collaborator repo was swept\n'
+	exit 1
+fi
+
+if ! grep -q '^other/maintain|' "$called_file"; then
+	printf 'FAIL maintain collaborator repo was not swept\n'
+	exit 1
+fi
+
+printf 'PASS stats routines require maintainer-equivalent access\n'

